src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -33,8 +33,12 @@ StaticDataCertificateProvider::StaticDataCertificateProvider(
     std::string root_certificate,
     grpc_core::PemKeyCertPairList pem_key_cert_pairs)
     : distributor_(MakeRefCounted<grpc_tls_certificate_distributor>()),
-      root_certificate_(std::move(root_certificate)),
-      pem_key_cert_pairs_(std::move(pem_key_cert_pairs)) {
+      root_certificate_(std::move(root_certificate)) {
+  if (IsKeyCertPairListValid(pem_key_cert_pairs).ok()) {
+    pem_key_cert_pairs_ = std::move(pem_key_cert_pairs);
```

Comment:
We could use ternary operator here
```
pem_key_cert_pairs_ = IsKeyCertPairListValid(pem_key_cert_pairs).ok() ? std::move(pem_key_cert_pairs) : {};
```

Reply:
I tried that and got `initializer list cannot be used on the right hand side of operator :` error so I just went with if-else.

Reply:
Similar situation: https://stackoverflow.com/questions/9982681/ternary-operator-c11-constructor-from-initializer-list